core/text/match/strlib.odin

```diff
@@ -0,0 +1,949 @@
+package strlib
```

Comment:
I'd recommend changing the name of this package to `package text_match` to minimize ABI issues.